libs/basics/src/iterators/types.ts

```diff
@@ -1035,6 +1063,37 @@ export interface AsyncIteratorPlus<T> extends AsyncIterable<T> {
     predicate: (item: T) => unknown
   ): Promise<[truthy: T[], falsy: T[]]>;
 
+  /**
+   * Reduces elements from `this` using `fn`. Consumes the entire contained
+   * iterable.
+   *
+   * @example
+   *
+   * ```ts
+   * const input = fs.createReadStream('file.txt', { encoding: 'utf8' });
+   * const longestLine = await lines(input).reduce((a, b) => a.length > b.length ? a : b);
```

Comment:
Maybe good to have an example that's not better implemented with another iter method (in this case `maxBy`)?